node_modules/oae-zencoder/lib/util.js

```diff
@@ -0,0 +1,129 @@
+            'amazons3-bucket': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'amazons3-bucket') || config['amazons3-bucket'],
+            'amazons3-access-key': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'amazons3-access-key') || config['amazons3-access-key'],
+            'amazons3-secret-key': ZencoderConfig.getValue(tenantAlias, 'zencoder', 'amazons3-secret-key') || config['amazons3-secret-key']
+        });
+    } else {
+        return config;
+    }
+};
+
+/**
+ * Create an encoding job in Zencoder.
+ *
+ * @param  {Object}     config              The default configuration
+ * @param  {String}     tenantAlias         The alias of the tenant whose configuration to use
+ * @param  {String}     path                The signed path to video/audio file
+ * @param  {Function}   callback            Standard callback function
+ * @param  {Error}      callback.err        An error that occurred, if any
+ * @param  {Response}   callback.response   The HTTP Response object
+ * @param  {String}     callback.body       The body of the response
+ */
+var createJob = module.exports.createJob = function(tenantAlias, path, contentId, revisionId, mime, callback) {
+    var config = getConfig(tenantAlias);
+    var isVideo = PreviewUtil.isVideo(mime);
+
+    var tenant = TenantsAPI.getTenant(tenantAlias);
+    var baseUrl = util.format('s3://s3-%s.amazonaws.com/%s/%s', config['amazons3-region'], config['amazons3-bucket'], encodeURIComponent(contentId));
+    var extension = (isVideo) ? 'mp4' : 'mp3';
+    var filename = util.format('%s.%s', revisionId.replace(/:/g, '-'), extension);
+
+    // @See https://app.zencoder.com/docs/api/encoding for details of Zencoder's encoding settings
+    var options = {
+        'input': util.format('https://%s%s', tenant.host, path),
+        'pass_through': util.format('%s/%s', contentId, filename), // We store the S3 bucket filepath so it can later be used to create a signed URL
+        'private': true,
+        'height': 600,
+        'output': {
+            'notifications': util.format('https://%s/api/zencoder/encodingNotification', tenant.host),
+            'base_url': baseUrl,
+            'filename': filename,
+            'format': extension
+        }
+    };
+
+    if (isVideo) {
+        options.output.thumbnails = [
+            {
+                'label': 'base',
+                'number': 1,
+                'base_url': baseUrl,
+                'height': PreviewConstants.SIZES.IMAGE.LARGE,
+                'width': PreviewConstants.SIZES.IMAGE.LARGE,
+                'filename': 'base',
+                'public': true
+            }
+        ];
+        if (config.watermark) {
+            options.output.watermarks = [
+                {
+                    'url': config.watermark,
+                    'opacity': 0.3,
+                    'width': '20%'
+                }
+            ];
+        }
```

Comment:
Do we have a use-case for the watermark? If not, maybe we should omit any logic and configuration associated to it for now or else we'll need to write tests with and without it to ensure we don't have any landmines.